src/main/java/org/opensearch/knn/index/codec/BasePerFieldKnnVectorsFormat.java

```diff
@@ -59,6 +86,46 @@ public KnnVectorsFormat getKnnVectorsFormatForField(final String field) {
         return formatSupplier.apply(maxConnections, beamWidth);
     }
 
+    private KnnVectorsFormat validateAndApplyQuantizedVectorsFormatForLuceneEngine(
+        final Map<String, Object> params,
+        final String field,
+        final int maxConnections,
+        final int beamWidth
+    ) {
+
+        if (params.get(METHOD_ENCODER_PARAMETER) == null) {
+            return null;
+        }
+
+        // Validate if the object is of type MethodComponentContext before casting it later
+        if (!(params.get(METHOD_ENCODER_PARAMETER) instanceof MethodComponentContext)) {
+            return null;
+        }
+        MethodComponentContext encoderMethodComponentContext = (MethodComponentContext) params.get(METHOD_ENCODER_PARAMETER);
+        if (!ENCODER_SQ.equals(encoderMethodComponentContext.getName())) {
+            return null;
+        }
+        Map<String, Object> sqEncoderParams = encoderMethodComponentContext.getParameters();
+        Float confidenceInterval = getConfidenceInterval(sqEncoderParams);
+        int bits = getBits(sqEncoderParams);
```

Comment:
Is there validation on this?

Reply:
The parameter validations are in Lucene.java class